decentraland/unity-explorer
# Feat: in world camera movement

## What does this PR change?

Implements transition to the in-world camera from different cinemachine states and control of its movement:

- Move Camera: Click and Drag
- Zoom: mouse wheel
- UP / DOWN: R / F
- FORWARD / BACKWARD / STRAFE LEFT / STRAFE RIGHT: W, S, A, D
- Rotate: MOUSE
- Nametags Switch: N
- Open/Close Camera: C
- Take a Picture: space
- LeftShift (hold): translation speed-up

I ported movement functionality from the old client. It is implemented via Cinemachine, but in this case uses actual player camera instead of creating new one. 
Includes bug fix for 90/-90 degree rotation that is presented in the old client. 

## How to test the changes?

1. Launch the explorer with the `--camera-reel` CLI argument
2. Press C to enter camera mode - check different edge cases for this  transitions (exploratory testing)
3. Use WASD, RF and mouse to move around
4. Verify you cannot pass through colliders (should be same as for your avatar)
5. Verify you camera movement is limited by around 16 metters around the player and never goes below zero ground level
6. Check the zoom via mouse scroll - play around with transitions to different states with zoom changed (exploratory testing)
7. Optionally: take a picture via Spacebar. You can check that picture by going to the old client (it should appear after a while in the CameraReel Gallery there)

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
